src/base/static/components/place-detail/index.js

```diff
@@ -210,6 +215,90 @@ class PlaceDetail extends Component {
     });
   }
 
+  onAdditionalData(action, payload) {
+    switch (action) {
+      case constants.ON_ADD_ATTACHMENT_ACTION:
+        this.attachments.push(payload);
+        break;
+      default:
+        console.error(
+          "Error: Unable to handle form field callback action:",
+          action
+        );
+        break;
+    }
+  }
+
+  onGeometryStyleChange(style) {
+    this.geometryStyle = style;
+  }
+
+  onEditorUpdate(fieldState) {
+    this.setState({ isEditFormSubmitting: true });
+
+    const attrs = fieldState
+      .filter(state => state.get(constants.FIELD_STATE_VALUE_KEY) !== null)
+      .map(state => state.get(constants.FIELD_STATE_VALUE_KEY))
+      .toJS();
+
+    if (fieldState.get(constants.GEOMETRY_PROPERTY_NAME)) {
+      attrs.style = this.geometryStyle;
+    }
+
+    // Replace image data in rich text fields with placeholders built from each
+    // image's name.
+    // TODO: This logic is better suited for the FormField component,
+    // perhaps in an onSave hook.
+    this.categoryConfig.fields
+      .filter(field => field.type === constants.RICH_TEXTAREA_FIELD_TYPENAME)
+      .forEach(field => {
+        attrs[field.name] = extractEmbeddedImages(attrs[field.name]);
+      });
+
+    this.attachments.forEach(attachment => {
+      this.props.model.attachmentCollection.add(attachment);
+    });
+
+    this.props.model.save(attrs, {
+      success: () => {
+        this.setState({ isEditFormSubmitting: false });
+        if (Backbone.history.fragment === Util.getUrl(this.props.model)) {
+          Backbone.history.loadUrl(Backbone.history.fragment);
+        } else {
+          this.props.router.navigate(Util.getUrl(this.props.model), {
+            trigger: true,
+            replace: true,
+          });
+        }
+      },
+      error: () => {
+        this.setState({ isEditFormSubmitting: false });
+        Util.log("USER", "place-editor", "fail-to-edit-place");
+      },
+    });
+  }
+
+  onEditorRemove() {
+    this.setState({ isEditFormSubmitting: true });
+    if (confirm(messages.confirmRemove)) {
```

Comment:
Yeah, it's similar to `alert()`--it's a a piece of built-in browser UI. Do you think we shouldn't use it here?

Reply:
Ah, interesting. I wonder if a `confirm` is the right UX here. See this article about using [warnings vs undo](http://alistapart.com/article/neveruseawarning)

Maybe it's out of scope, but perhaps an "undo" option would be better?

Reply:
Yes I think we'd benefit from an undo feature in the editor (using `confirm()` was just a quick fix). I'm not sure if we should add an undo feature to this PR though. I could see that as a separate issue/PR down the road though.

Reply:
What does `confirm` fix? It seems to do more harm than good, because it creates a pop up that freeze the user's screen, adds more friction, and potentially confuses the user.

I'd prefer to just remove it, and we can add an "undo" option at a later time. But I'll trust whatever you decide - you have more familiarity with the feature and the UX.

Reply:
`confirm()` prevents an accidental (single) click on a button that would otherwise permanently remove a post that the user is editing. For that reason I'd argue to keep it in place until we have a proper undo feature.

Undo is definitely the better UX pattern here, but I think an undo feature is complex (e.g. where does the undo UI go after a post is removed? What is the scope of the undo? Does it undo edits? etc.). We should think through these things, it just seems like they're out of scope in this PR.